IP routing fails with flannel interface

Some interfaces such as flannel have `.` in the name (`flannel.1`) and the associated `/proc/sys/net/ipv4/conf/` is actually called `flannel.1` rather than `flannel/1`.

This causes issues when Netbird client tries to set up server advertised routes.

The client shows following error:
```
2025-08-23T21:07:50Z ERRO client/internal/routemanager/systemops/systemops_linux.go:124: Error setting up sysctl: 1 error occurred:
	* read sysctl net.ipv4.conf.flannel.1.rp_filter: open /proc/sys/net/ipv4/conf/flannel/1/rp_filter: no such file or directory
```

I have been able to trace this to the following code that is responsible for resolving the `rp_filter` in directory named after the interface: [link](https://github.com/netbirdio/netbird/blob/5e273c121ac207ed36ca5367ff70eeeb056cc6cd/client/internal/routemanager/sysctl/sysctl_linux.go#L73)

My understanding is that this issue is causing client to fail configuring advanced routing, which as a result causes routes advertised by management/server not to work.

**My environment**
I am testing in kubernetes RKE2 environment with latest Netbird client running in a container/pod with `privileged` access with `hostNetwork: true` enabled. The host has a flannel interface `flannel.1`, which is part of the kubernetes stack.

--- a/client/internal/routemanager/sysctl/sysctl_linux.go
+++ b/client/internal/routemanager/sysctl/sysctl_linux.go
@@ -20,7 +20,8 @@ const (
 	rpFilterPath          = "net.ipv4.conf.all.rp_filter"
 	rpFilterInterfacePath = "net.ipv4.conf.%s.rp_filter"
 	srcValidMarkPath      = "net.ipv4.conf.all.src_valid_mark"
-	dotEscape             = "__DOT__"
+	percentEscape         = "%25"
+	dotEscape             = "%2E"
 )
 
 type iface interface {
@@ -57,8 +58,10 @@ func Setup(wgIface iface) (map[string]int, error) {
 			continue
 		}
 
-		// Escape dots in the interface name (e.g. flannel.1 -> flannel__DOT__1)
-		safeName := strings.ReplaceAll(intf.Name, ".", dotEscape)
+		// Escape '%' and '.' so they survive the dot-to-slash conversion in Set()
+		safeName := strings.ReplaceAll(intf.Name, "%", percentEscape)
+		safeName = strings.ReplaceAll(safeName, ".", dotEscape)
+
 		i := fmt.Sprintf(rpFilterInterfacePath, safeName)
 		oldVal, err := Set(i, 2, true)
 		if err != nil {
@@ -74,8 +77,9 @@ func Setup(wgIface iface) (map[string]int, error) {
 // Set sets a sysctl configuration, if onlyIfOne is true it will only set the new value if it's set to 1
 func Set(key string, desiredValue int, onlyIfOne bool) (int, error) {
 	path := strings.ReplaceAll(key, ".", "/")
-	// Unescape dot in interface name (e.g. flannel__DOT__1 -> flannel.1)
+	// Unescape interface dots and percent signs
 	path = strings.ReplaceAll(path, dotEscape, ".")
+	path = strings.ReplaceAll(path, percentEscape, "%")
 	path = fmt.Sprintf("/proc/sys/%s", path)
 	currentValue, err := os.ReadFile(path)
 	if err != nil {

--- a/client/internal/routemanager/sysctl/sysctl_linux.go
+++ b/client/internal/routemanager/sysctl/sysctl_linux.go
@@ -20,6 +20,7 @@ const (
 	rpFilterPath          = "net.ipv4.conf.all.rp_filter"
 	rpFilterInterfacePath = "net.ipv4.conf.%s.rp_filter"
 	srcValidMarkPath      = "net.ipv4.conf.all.src_valid_mark"
+	dotEscape             = "__DOT__"
 )
 
 type iface interface {
@@ -56,7 +57,9 @@ func Setup(wgIface iface) (map[string]int, error) {
 			continue
 		}
 
-		i := fmt.Sprintf(rpFilterInterfacePath, intf.Name)
+		// Escape dots in the interface name (e.g. flannel.1 -> flannel__DOT__1)
+		safeName := strings.ReplaceAll(intf.Name, ".", dotEscape)
+		i := fmt.Sprintf(rpFilterInterfacePath, safeName)
 		oldVal, err := Set(i, 2, true)
 		if err != nil {
 			result = multierror.Append(result, err)
@@ -70,7 +73,10 @@ func Setup(wgIface iface) (map[string]int, error) {
 
 // Set sets a sysctl configuration, if onlyIfOne is true it will only set the new value if it's set to 1
 func Set(key string, desiredValue int, onlyIfOne bool) (int, error) {
-	path := fmt.Sprintf("/proc/sys/%s", strings.ReplaceAll(key, ".", "/"))
+	path := strings.ReplaceAll(key, ".", "/")
+	// Unescape dot in interface name (e.g. flannel__DOT__1 -> flannel.1)
+	path = strings.ReplaceAll(path, dotEscape, ".")
+	path = fmt.Sprintf("/proc/sys/%s", path)
 	currentValue, err := os.ReadFile(path)
 	if err != nil {
 		return -1, fmt.Errorf("read sysctl %s: %w", key, err)

